### DISCUSSION TITLE
How can i create a simple chat agent from my existing application frontend to Dify through API?

### DISCUSSION THREAD

**Discussion by outsurface** [langgenius/dify#9979] [General]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] [FOR CHINESE USERS] 请务必使用英文提交 Issue，否则会被关闭。谢谢！:）
- [X] Please do not modify this template :) and fill in all the required fields.

### Content

I'm using Dify to have my chat bots but I'm working for a company who wants to customize or have more chatbots i want to design a simple form & it should  create a chatbot in Dify & further i have API to load the chatbot in the application frontend.

Note : I don't want to give the Dify self hosted version access as i have more chatbots there which i don't want them to access.

**Comment by outsurface** [comment]
As i can see there is a DSL file import option with URL is also available can i make that file in my application & send it to Dify through some API & can i get the app id to further integrate it?

**Comment by julianodev** [comment]
Any news?

**Comment by wahyu-saputra-personal** [comment]
Also interested in this. My projects involves creating many chatbots each of my company's agents. I am thinking of:
1. Able to create new chatbot from the template DSL file via api, so automate the onboarding process without going through the UI 
2. Able to "update" or reimport the template via api, so if I make changes to the template it can easily propagated without doing it one by one

**Comment by luiszxcv** [comment]
Up

**Comment by yaohwang** [comment]
create api to make chatbot or agent, then use them with chatbot's api (with chatbot id etc.). does now support that?

**Comment by biofer** [comment]
Same request

**Comment by jakubscholtz-inda** [comment]
I would also like to know if there is a way to programatically upload/update the dsl file.

**Comment by o-smallit** [comment]
Same request. I'd like to create a chatbot via API.

**Reply by Eizokie** [reply]
Any update? Can you help me bro if you know how to do it?
